processing/src/main/java/io/druid/query/filter/LongValueMatcherColumnSelectorStrategy.java

```diff
@@ -28,7 +28,10 @@
   @Override
   public ValueMatcher makeValueMatcher(final LongColumnSelector selector, final String value)
   {
-    final Long matchVal = DimensionHandlerUtils.convertObjectToLong(value);
+    Long matchVal = DimensionHandlerUtils.convertObjectToLong(value);
+    if (matchVal == null) {
+      matchVal = DimensionHandlerUtils.getIntegralFromFloatString(value);
```

Comment:
Moved the call inside convertObjectToLong, also got rid of unnecessary try-catches within convertObjectToLong and convertObjectToFloat